### PR TITLE
Remove Python 2 compatibility

### PR DESCRIPTION
**Pull request by mdickinson**
This mostly mechanical search-and-replace PR removes the majority of the Python 2 compatibility code from the Traits codebase.

Known remnants:

- `ctraits.c` and the supporting include file `py2to3.h`
- `_py2to3.py` for `contextlib.nested`: fixed in #639

### CODE REVIEW

File: traits/trait_handlers.py [hunk 1]
```diff
@@ -2956,7 +2928,7 @@ def __deepcopy__(self, memo):
             self.trait,
             lambda: None,
             self.name,
-            dict([copy.deepcopy(x, memo) for x in six.iteritems(self)]),
+            dict([copy.deepcopy(x, memo) for x in self.items()]),
```
Comment (corranwebster) [inline traits/trait_handlers.py]: Drive-by fix: is the intermediate list actually needed here?

Comment (mdickinson) [reply traits/trait_handlers.py]: Very likely not.
```suggestion
            dict(copy.deepcopy(x, memo) for x in self.items()),
```
